homeassistant/components/bryant_evolution/climate.py

```diff
@@ -0,0 +1,241 @@
+"""Support for Bryant Evolution HVAC systems."""
+
+from datetime import timedelta
+import logging
+from typing import Any
+
+from evolutionhttp import BryantEvolutionClient
+
+from homeassistant.components.climate import (
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACAction,
+    HVACMode,
+)
+from homeassistant.const import CONF_HOST, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import BryantEvolutionConfigEntry
+from .const import CONF_SYSTEM_ID, CONF_ZONE_ID, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+SCAN_INTERVAL = timedelta(seconds=60)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: BryantEvolutionConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up a config entry."""
+    client = config_entry.runtime_data
+    climate = BryantEvolutionClimate(
+        config_entry.data[CONF_HOST],
+        config_entry.data[CONF_SYSTEM_ID],
+        config_entry.data[CONF_ZONE_ID],
+        client,
+    )
+    async_add_entities([climate], update_before_add=True)
+
+
+class BryantEvolutionClimate(ClimateEntity):
+    """ClimateEntity for Bryant Evolution HVAC systems.
+
+    Design note: this class updates using polling. However, polling
+    is very slow (~1500 ms / parameter). To improve the user
+    experience on updates, we also locally update this instance and
+    call async_write_ha_state as well.
+    """
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self: ClimateEntity,
+        host: str,
+        system_id: int,
+        zone_id: int,
+        client: BryantEvolutionClient,
+    ) -> None:
+        """Initialize an entity from parts."""
+        self._client = client
+        self._attr_name = None
+        self._attr_temperature_unit = UnitOfTemperature.FAHRENHEIT
+        self._attr_supported_features = (
+            ClimateEntityFeature.TARGET_TEMPERATURE
+            | ClimateEntityFeature.TARGET_TEMPERATURE_RANGE
+            | ClimateEntityFeature.FAN_MODE
+            | ClimateEntityFeature.TURN_ON
+            | ClimateEntityFeature.TURN_OFF
+        )
+        self._attr_hvac_modes = [
+            HVACMode.HEAT,
+            HVACMode.COOL,
+            HVACMode.HEAT_COOL,
+            HVACMode.OFF,
+        ]
+        self._attr_fan_modes = ["AUTO", "LOW", "MED", "HIGH"]
+        self._enable_turn_on_off_backwards_compatibility = False
+        self._attr_unique_id = f"bryant_evolution_{host}_{system_id}_{zone_id}"
```

Comment:
the unique_id is unique per domain per platform, so there is no need to prefix it